### PR TITLE
[KOA-4846] Fixed white spaces selection in calendar component

### PR DESCRIPTION
**Pull request by anambl**
|Before https://backpack.github.io/storybook/?path=/story/bpk-component-scrollable-calendar--scrollable-calendar-range |After https://backpack.github.io/storybook-prs/2296/?path=/story/bpk-component-scrollable-calendar--scrollable-calendar-range |
|-|-|
|image|image|

Remember to include the following changes:

- [x] `UNRELEASED.md` (See [How to write a good changelog entry](https://github.com/Skyscanner/backpack/blob/main/CHANGELOG_FORMAT.md))
- [ ] `README.md`
- [ ] Tests
- [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

### CODE REVIEW

**Review by olliecurtis — CHANGES_REQUESTED** [review]
Just a couple of minior checks we should just verify to avoid unexpected crashes going on during the calendar operation

**Comment by olliecurtis** [comment]
Also sorry I didn't mean to use request changes but comment instead 😂

File: packages/bpk-component-calendar/src/Week.js [hunk 1]
```diff
@@ -83,9 +85,24 @@ function getSelectedDate(date, selectionConfiguration) {
...
+function getSelectionType(
+  date,
+  selectionConfiguration,
+  formatDateFull,
+  month,
+  weekStartsOn,
+  ignoreOutsideDate,
+) {
+  const { startDate, endDate } = selectionConfiguration;
+  const sameStartDay = isSameDay(date, startDate);
```
[the remark this answers is not in the document]
Comment (anambl) [reply packages/bpk-component-calendar/src/Week.js:102]: good point! I'll check this!

[the remark this answers is not in the document]
Comment (anambl) [reply packages/bpk-component-calendar/src/Week.js:102]: It seems like it should be alright as the function returns false if one or both dates are null 🤔

**Review by olliecurtis — APPROVED** [review]
This is amazing!

**Pull request merged**